### ISSUE TITLE
Current structure of frontends - hard to independently test each

### ISSUE THREAD

**Issue by awilfox** [supergameherm/supergameherm#29]
The fact that frontends are specified in the generated config.h is making it very hard to test changes to one frontend without affecting all the others.

For instance, I was testing changes using the SDL2 frontend and broke it badly.  I went to the null frontend to try and see what the result would be using it, and it pulled in SDL2 headers that the null frontend does not need.  Since I was writing these changes on Mac OS X where SDL2 headers live in frameworks, this resulted in "SDL2/SDL.h does not exist" errors because that framework is, for obvious reasons, not included in the null frontend's build flags.

Basically, the current structure requires all frontends to be buildable at the same time, which is not quite feasible on "extravagant" platforms like Mac OS X or Windows where SDL2 does not live in the system include path and include paths are not global.

I'm not sure exactly how to fix this, hence this is tagged "help wanted".  I would love to hear ideas on how to fix this while keeping our portability and build system sane.

**Comment by Elizafox** [comment]
I am not quite sure I understand what you mean... I think what you're really trying to say is you want each frontend to be a subproject? I don't know how to go about this at all.

**Comment by awilfox** [comment]
*Please note GitHub no longer supports this version of Safari.*  Fuck you too, GitHub.  Leopard is like the LTS of the Mac OS X world; suck it up.

Ahem.  Sorry.

```c
#define HAVE_SDL2
```

is breaking the sgherm-null target in Xcode in the following manner:

```

/Users/awilcox/supergameherm-master/include/mbc.h:107: warning: ISO C doesn't support unnamed structs/unions
In file included from /Users/awilcox/supergameherm-master/include/frontends/sdl2/frontend.h:4,
                 from /Users/awilcox/supergameherm-master/include/frontend.h:79,
                 from /Users/awilcox/supergameherm-master/include/sgherm.h:15,
                 from /Users/awilcox/supergameherm-master/src/frontends/null/main.c:3:
/Users/awilcox/supergameherm-master/include/frontends/sdl2/sdl_inc.h:15:22: error: SDL2/SDL.h: No such file or directory
```

Note that this is because ```-framework SDL2``` is not being passed on the command line.  And it should *not* be passed on the command line for the -null or -gdi or -caca or any other back end.

**Comment by Elizafox** [comment]
I think I might have a solution.
